File: src/VBNetFx/FacessoData/SQLServer/SPAccess_Main.vb
Imports Activedev
Imports System.Data
Imports System.Data.SqlClient
Imports System.Data.Common

Partial Public NotInheritable Class SPAccess

    Private Shared myInstance As SPAccess

    Private Sub New()
        'myInstance = Nothing
    End Sub

    Public Shared Function GetInstance() As SPAccess
        If myInstance Is Nothing Then
            If FacessoGeneric.FacessoLicenseInfo.IsLicensed Then
                myInstance = New SPAccess()
            Else
                Dim up As New FacessoLicenseViolationException(My.Resources.Sp_Main_FailedInstancingDueToLicence, Nothing)
                Throw up
            End If
        End If
        Return myInstance
    End Function

    Public Function GetOpenedConnectionSafely() As SqlConnection
        Dim locConnection As New SqlConnection(FacessoGeneric.SQLConnectionString)
        Try
            locConnection.Open()
        Catch ex As Exception
            'Anwendungs-Exception auslösen
            Dim locString As String = My.Resources.Sp_Main_OpenFacessoConnectionFailed + vbNewLine + vbNewLine
            locString &= ex.Message
            Dim up As New FacessoSqlDbException(locString, ex)
            Throw up
            Return Nothing
        End Try
        Return locConnection
    End Function

    Public ReadOnly Property SQLConnectionString() As String
        Get
            Return FacessoGeneric.SQLConnectionString
        End Get
    End Property

    Public Sub DeleteDataForOleDbImport(ByVal IDSubsidiary As Integer)

        Dim locConnection As SqlConnection = GetOpenedConnectionSafely()
        If locConnection Is Nothing Then
            Return
        End If
        Using locConnection
            Dim locCmd As New SqlCommand("DeleteDataForOleDbImport", locConnection)
            locCmd.CommandType = CommandType.StoredProcedure
            With locCmd.Parameters
                .Add("@IDSubsidiary", SqlDbType.Int).Value = IDSubsidiary
            End With
            locCmd.ExecuteReader()
        End Using
    End Sub
End Class

File: src/MixNetFx/FacessoData/SQLServer/SPAccess_Main.cs
using System;
using System.Data;
using System.Data.SqlClient;
using ActiveDev;
using Facesso;
using Facesso.Data.My.Resources;

namespace Facesso.Data
{
    public sealed partial class SPAccess
    {
        private static SPAccess myInstance;

        private SPAccess() { }

        public static SPAccess GetInstance()
        {
            if (myInstance == null)
            {
                if (FacessoGeneric.FacessoLicenseInfo.IsLicensed())
                {
                    myInstance = new SPAccess();
                }
                else
                {
                    throw new FacessoLicenseViolationException(Resources.Sp_Main_FailedInstancingDueToLicence, null);
                }
            }
            return myInstance;
        }

        public SqlConnection GetOpenedConnectionSafely()
        {
            var locConnection = new SqlConnection(FacessoGeneric.SQLConnectionString);
            try
            {
                locConnection.Open();
            }
            catch (Exception ex)
            {
                string locString = Resources.Sp_Main_OpenFacessoConnectionFailed + Environment.NewLine + Environment.NewLine;
                locString += ex.Message;
                throw new FacessoSqlDbException(locString, ex);
            }
            return locConnection;
        }

        public string SQLConnectionString => FacessoGeneric.SQLConnectionString;

        public void DeleteDataForOleDbImport(int idSubsidiary)
        {
            SqlConnection locConnection = GetOpenedConnectionSafely();
            if (locConnection == null) return;
            using (locConnection)
            {
                var locCmd = new SqlCommand("DeleteDataForOleDbImport", locConnection);
                locCmd.CommandType = CommandType.StoredProcedure;
                locCmd.Parameters.Add("@IDSubsidiary", SqlDbType.Int).Value = idSubsidiary;
                locCmd.ExecuteReader();
            }
        }
    }
}
